substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -1853,7 +1853,17 @@ impl<T: Config> Pallet<T> {
 
 	fn check_ledgers() -> Result<(), TryRuntimeError> {
 		Bonded::<T>::iter()
-			.map(|(_, ctrl)| Self::ensure_ledger_consistent(ctrl))
+			.map(|(stash, ctrl)| {
+				// `ledger.controller` is never stored in raw storage.
+				let raw = Ledger::<T>::get(stash).unwrap_or_else(|| {
+					Ledger::<T>::get(ctrl.clone())
+						.expect("try_check: bonded stash/ctrl does not have an associated ledger")
```

Comment:
Should we just log an error or panic? Since try state checks are mostly caused by older issues, we may not want CI to fail for these errors?